### PR TITLE
[aot] Size optimizations.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
ty

File: mono/mini/aot-runtime.c [hunk 1]
```diff
@@ -436,26 +436,26 @@ decode_generic_inst (MonoAotModule *module, guint8 *buf, guint8 **endbuf, MonoEr
...
+
+	/* Either the class_inst or method_inst offset */
+	flags = decode_value (p, &p);
 
-	p2 = p;
-	argc = decode_value (p, &p);
-	if (argc) {
-		p = p2;
-		ctx->class_inst = decode_generic_inst (module, p, &p, error);
+	if (flags & 1) {
```
Comment (lewurm) [inline mono/mini/aot-runtime.c]: please use some enum instead of `1` and `2` for `flags`

**Pull request merged**